src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -37,57 +39,404 @@ const ActorInit En_Cow_InitVars = {
+        this->actor.flags &= ~0x10000;
+        func_801477B4(globalCtx);
+        this->actionFunc = EnCow_GiveMilkWait;
+        func_800B8A1C(&this->actor, globalCtx, GI_MILK, 10000.0f, 100.0f);
+    }
+}
+
+void EnCow_CheckForEmptyBottle(EnCow* this, GlobalContext* globalCtx) {
+    if ((func_80152498(&globalCtx->msgCtx) == 5) && func_80147624(globalCtx)) {
+        if (func_80114E90()) {
+            func_80151938(globalCtx, 0x32C9);
+            this->actionFunc = EnCow_GiveMilk;
+        } else {
+            func_80151938(globalCtx, 0x32CA);
+            this->actionFunc = EnCow_TalkEnd;
+        }
+    }
+}
+
+void EnCow_Talk(EnCow* this, GlobalContext* globalCtx) {
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        if (this->actor.textId == 0x32C8) {
+            this->actionFunc = EnCow_CheckForEmptyBottle;
+        } else if (this->actor.textId == 0x32C9) {
+            this->actionFunc = EnCow_GiveMilk;
+        } else {
+            this->actionFunc = EnCow_TalkEnd;
+        }
+    } else {
+        this->actor.flags |= 0x10000;
+        func_800B8614(&this->actor, globalCtx, 170.0f);
+        this->actor.textId = 0x32C8; //! @bug textId is reset to this no matter the intial value
+    }
+
+    EnCow_UpdateAnimation(this, globalCtx);
+}
+
+void EnCow_Idle(EnCow* this, GlobalContext* globalCtx) {
+    if ((globalCtx->msgCtx.unk1202A == 0) || (globalCtx->msgCtx.unk1202A == 4)) {
+        if (D_801BDAA4 != 0) {
+            if (this->flags & EN_COW_FLAG_WONT_GIVE_MILK) {
+                this->flags &= ~EN_COW_FLAG_WONT_GIVE_MILK;
+                D_801BDAA4 = 0;
+            } else if ((this->actor.xzDistToPlayer < 150.0f) &&
+                       ABS_ALT((s16)(this->actor.yawTowardsPlayer - this->actor.shape.rot.y)) < 25000) {
+                D_801BDAA4 = 0;
+                this->actionFunc = EnCow_Talk;
+                this->actor.flags |= 0x10000;
+                func_800B8614(&this->actor, globalCtx, 170.0f);
+                this->actor.textId = 0x32C8;
+
+                EnCow_UpdateAnimation(this, globalCtx);
+                return;
+            } else {
+                this->flags |= EN_COW_FLAG_WONT_GIVE_MILK;
+            }
+        } else {
+            this->flags &= ~EN_COW_FLAG_WONT_GIVE_MILK;
+        }
+    }
+
+    if (this->actor.xzDistToPlayer < 150.0f &&
+        ABS_ALT((s16)(this->actor.yawTowardsPlayer - this->actor.shape.rot.y)) < 25000) {
+        if (func_801A5100() == 4) {
+            if (!(gSaveContext.weekEventReg[87] & 1)) {
+                gSaveContext.weekEventReg[87] |= 1;
+                if (func_80114E90() != 0) {
+                    this->actor.textId = 0x32C9;
+                } else {
+                    this->actor.textId = 0x32CA;
+                }
+                this->actor.flags |= 0x10000;
+                func_800B8614(&this->actor, globalCtx, 170.0f);
+                this->actionFunc = EnCow_Talk;
+            }
+        } else {
```

Comment:
`func_801A5100` is actually related to the voice recognition unit (VRU), and checks if a certain word has been said! Meaning this whole block of code is VRU exclusive!

Reply:
Oh that's neat! I was wondering what that was since there was no documentation for what that `weekEventReg` flag did. I guess they were intending for you to speak to get free milk?

Implementing the VRU functionality would be a cool mod.

Reply:
You mean like this mod: 👀
https://www.youtube.com/watch?v=VxK8XqYecSg

Reply:
Yep! And he's indeed getting milk in it.